### PR TITLE
Switch to bellpepper

### PR DESCRIPTION
**Pull request by huitseeker**
Hides bellperson / bellpepper behind a "bellbar" re-export which nature is controlled by the "groth16" feature.
Breaking change: those who want to stay on bellperson have to activate this non-default feature.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -51,15 +51,16 @@ jobs:
...
-        # TODO: --all-features
         run: cargo build --benches --release
       - name: cargo test
-        # TODO: --all-features
         run: |
           cargo nextest run --release
       - name: Doctests
         run: |
           cargo test --doc
+      - name: cargo test (groth16)
```
Comment (porcuquine) [inline .github/workflows/rust.yml:61]: I wouldn't call this feature `groth16`. It's specifically about which library (`bellpepper` vs `bellperson`) provides the `ConstraintSystem` trait used for circuit synthesis. For example, Nova also currently uses `bellperson` despite having no relationship to Groth16,

File: src/lib.rs [hunk 2]
```diff
@@ -39,6 +39,12 @@ compile_error!("The `strengthened` feature needs the `cuda` and/or `opencl` feat
 ))]
 compile_error!("The `cuda` and `opencl` features need the `bls` and/or `pasta` feature to be set");
 
+#[cfg(all(any(feature = "cuda", feature = "opencl"), not(feature = "groth16")))]
+compile_error!("The `cuda` and `opencl` features need the `groth16` feature to be set");
```
Comment (porcuquine) [inline src/lib.rs:43]: Do we *need* to forbid GPU use with `bellpepper`? That seems sub-optimal and would be worth working around if possible.

Comment (porcuquine) [reply src/lib.rs:43]: Neptune uses GPU for non-circuit hashing, and this configuration would seem to disallow that.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (huitseeker) [reply .github/workflows/rust.yml]: Rename to bellpepper!

File: src/lib.rs (hunk 2 above)
Comment (huitseeker) [reply src/lib.rs]: Yes, my mistake, I fixed it, there and in CI.

**Comment by huitseeker** [comment]
Closing in favor of #217

**Pull request closed without merging**